engine/src/main/java/org/hibernate/search/engine/metadata/impl/DocumentFieldMetadata.java

```diff
@@ -216,13 +226,14 @@ public String toString() {
 
 		public Builder(BackReference<TypeMetadata> sourceType,
 				BackReference<PropertyMetadata> sourceProperty,
-				String absoluteName,
+				String absoluteName, String relativeName,
```

Comment:
It doesn't feel weird to me, but I agree enclosing it in a separate object would be cleaner. I think we could leverage the `PathContext` to generate those paths, too. 
I'll work on that.